src/electrum_extended_pub_key.rs

```diff
@@ -53,6 +53,36 @@ impl ElectrumExtendedPubKey {
     pub fn xpub(&self) -> &ExtendedPubKey {
         &self.xpub
     }
+
+    /// converts to electrum format
+    pub fn electrum_xpub(&self) -> Result<String, String> {
+        let data = match (self.xpub.network, self.kind.as_str()) {
+            (Network::Testnet, "pkh") => [0x04u8, 0x35, 0x87, 0xcf], // tpub
```

Comment:
I think we need to add constants for all the bytes vec like `[0x04u8, 0x35, 0x87, 0xcf]` and also for the strings like `"pkh"` since we use it in multiple places

Reply:
> I think we need to add constants for all the bytes vec like `[0x04u8, 0x35, 0x87, 0xcf]` and also for the strings like `"pkh"` since we use it in multiple places

Yes, that would be cool. I found https://docs.rs/bimap/latest/bimap/ to store the values. But I didn't see how I could initialize a const bimap. Here is what I tried:
`type SentinelMap = BiMap<[u8;4], (Network, String)>;
const SENTINELS: SentinelMap = initialize_sentinels();

const fn initialize_sentinels() -> SentinelMap {
    let mut sentinels = BiMap::new();

    sentinels.insert([0x04u8, 0x35, 0x87, 0xcf], (Network::Testnet, "pkh".to_string())); // tpub
    sentinels.insert([0x04u8, 0x4a, 0x52, 0x62], (Network::Testnet, "sh(wpkh".to_string())); // upub
    sentinels.insert([0x02u8, 0x42, 0x89, 0xef], (Network::Testnet, "sh(wsh".to_string())); // Upub
    sentinels.insert([0x04u8, 0x5f, 0x1c, 0xf6], (Network::Testnet, "wpkh".to_string())); // vpub
    sentinels.insert([0x02u8, 0x57, 0x54, 0x83], (Network::Testnet, "wsh".to_string())); // Vpub
    sentinels.insert([0x04u8, 0x88, 0xB2, 0x1E], (Network::Bitcoin, "pkh".to_string())); // xpub
    sentinels.insert([0x04u8, 0x9d, 0x7c, 0xb2], (Network::Bitcoin, "sh(wpkh".to_string())); // ypub
    sentinels.insert([0x02u8, 0x95, 0xb4, 0x3f], (Network::Bitcoin, "sh(wsh".to_string())); // Ypub
    sentinels.insert([0x04u8, 0xb2, 0x47, 0x46], (Network::Bitcoin, "wpkh".to_string())); // zpub
    sentinels.insert([0x02u8, 0xaa, 0x7e, 0xd3], (Network::Bitcoin, "wsh".to_string())); // Zpub

    sentinels
}`
I am pretty sure, something like this would work in C++
So, would you declare just a couple of simple consts, without fixing the relationships in a central place?

Reply:
I ended up just calling the function in places where I would use the const bimap. That way the container is constructed more times than strictly necessary, but I don't think it's too bad.

Reply:
I don't think performances really matter here so it's fine initializing it more times.

I would have preferred to avoid using an additional dep, eg. using a `Vec` and searching it by simply iterating